documentation/design/user-stories/shared-infra.adoc

```diff
@@ -229,75 +280,297 @@ metadata:
+an unrestricted plan will be used as default.
+
+* If the reference does not resolve, the infra status will contain a message describing the issue.
+* A warning status will be set if the union of the `MessagingInfra` and the `MessagingPlan` selector is a disjoint set.
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingInfra
+metadata:
+  name: myinfra
+  namespace: infraspace
+spec:
+  defaultPlanRef:
+    name: myplan
+    namespace: planspace
+```
 
 === Specify default address plans for messaging tenants
 
-=== CRUDLW an instance of messaging infrastructure shared between multiple messaging tenants
+TODO
+
+=== Specify default endpoints and cert providers to be created for messaging tenants
+
+Default endpoint configuration is specified in the `MessagingPlan` under the `default` field.
+
+* The controller creates one `MessagingEndpoint` for each entry in the plan list
+* The schema for each entry corresponds to the `spec` object in `MessagingEndpoint` with the following exceptions:
+  * The `host` field is ignored and set to defaults
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingPlan
+metadata:
+  name: myplan
+  namespace: planspace
+spec:
+  default:
+    endpoints:
+     - route: {}
+       protocols:
+       - AMQPS
+       tls:
+         openshift: {}
+     - cluster: {}
+        # Define which protocols are supported by the default endpoint
+       protocols:
+       - AMQP
+```
+
+=== Restrict which endpoint types and cert providers can be used for messaging tenants
+
+Restricting allowed endpoint types and cert providers allow the service admin to control
+exposure of messaging endpoints in a cluster. The `restrict.endpoints` field is used to restrict
+which endpoint types and which protocols are allowed.
+
+* `MessagingEndpoint` resources for tenants referencing this plan must exactly match one of the entries.
+* The schema for each entry corresponds to the `spec` object in `MessagingEndpoint` with the following exceptions:
+  * The `host` field is ignored and set to defaults
+
+The following example will enforce `route` endpoints to use a specific certificate and require `cluster` endpoints to use unencrypted protocols.
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingPlan
+metadata:
+  name: myplan
+  namespace: planspace
+spec:
+  restrict:
+    endpoints:
+     - route: {}
+       protocols:
+       - AMQPS
+       tls:
+         external:
+           cert:
+             valueFromSecret:
+               name: external-certs
+     - cluster: {}
+        # Define which protocols are supported by the default endpoint
+       protocols:
+       - AMQP
+```
+
 
 === Restrict how much memory and storage is consumed per messaging tenant
 
+Create the following `MessagingPlan` to enforce limits:
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingPlan
+metadata:
+  name: plan1
+  namespace: planspace
+spec:
+  broker:
```

Comment:
should there be a restrict object here?

Reply:
I'm not sure, there are kind of conflicts of interest here between the desire to provide kubernetes-type resource specifications, which have both requests and limits, and restrictions which are kind of just restrictions.

Need to think some more about that.